lib/src/widgets/type_indicator_widget.dart

```diff
@@ -0,0 +1,247 @@
+
+import '../../chat_ui.dart';
+
+class TypingIndicator extends StatefulWidget {
+  const TypingIndicator({
+    Key? key,
+    this.showIndicator = false,
+    this.profilePic,
+    this.chatBubbleConfig,
+    this.typeIndicatorConfig,
+  }) : super(key: key);
+
+  final bool showIndicator;
+  final String? profilePic;
+  final ChatBubble? chatBubbleConfig;
+  final TypeIndicatorConfiguration? typeIndicatorConfig;
+
+  @override
+  _TypingIndicatorState createState() => _TypingIndicatorState();
+}
+
+class _TypingIndicatorState extends State<TypingIndicator>
+    with TickerProviderStateMixin {
+  late AnimationController _appearanceController;
+
+  late Animation<double> _indicatorSpaceAnimation;
+
+  late Animation<double> _largeBubbleAnimation;
+
+  late AnimationController _repeatingController;
+  final List<Interval> _dotIntervals = const [
+    Interval(0.25, 0.8),
+    Interval(0.35, 0.9),
+    Interval(0.45, 1.0),
+  ];
+
+  final List<AnimationController> _jumpControllers = [];
+  final List<Animation> _jumpAnimations = [];
+
+  ChatBubble? get chatBubbleConfig => widget.chatBubbleConfig;
+
+  double get indicatorSize => widget.typeIndicatorConfig?.indicatorSize ?? 10;
+
+  double get indicatorSpacing =>
+      widget.typeIndicatorConfig?.indicatorSpacing ?? 4;
+
+  Color? get flashingCircleDarkColor =>
+      widget.typeIndicatorConfig?.flashingCircleDarkColor ??
+      const Color(0xFF939497);
+
+  Color? get flashingCircleBrightColor =>
+      widget.typeIndicatorConfig?.flashingCircleBrightColor ??
+      const Color(0xFFadacb0);
+
+  @override
+  void initState() {
+    super.initState();
+
+    _appearanceController = AnimationController(
+      vsync: this,
+    )..addListener(() {
+        setState(() {});
+      });
+
+    _indicatorSpaceAnimation = CurvedAnimation(
+      parent: _appearanceController,
+      curve: const Interval(0.0, 0.4, curve: Curves.easeOut),
+      reverseCurve: const Interval(0.0, 1.0, curve: Curves.easeOut),
+    ).drive(Tween<double>(
+      begin: 0.0,
+      end: 60.0,
+    ));
+
+    _largeBubbleAnimation = CurvedAnimation(
+      parent: _appearanceController,
+      curve: const Interval(0.3, 1.0, curve: Curves.elasticOut),
+      reverseCurve: const Interval(0.5, 1.0, curve: Curves.easeOut),
+    );
+
+    _repeatingController = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 1500),
+    );
+
+    for (int i = 0; i < 3; i++) {
+      _jumpControllers.add(AnimationController(
+        vsync: this,
+        duration: const Duration(milliseconds: 500),
+        reverseDuration: const Duration(milliseconds: 500),
+      ));
+      _jumpAnimations.add(CurvedAnimation(
+              parent: _jumpControllers[i],
+              curve: Interval((0.2 * i), 0.7, curve: Curves.easeOutSine),
+              reverseCurve: Interval((0.2 * i), 0.7, curve: Curves.easeOut))
+          .drive(Tween<double>(begin: 0, end: 10)));
+    }
```

Comment:
create separate function for all this so initState would look clean

Reply:
Updated